lib/simple_form/components/labels.rb

```diff
@@ -50,7 +53,7 @@ def label_html_options
           label_options[:for] = options[:input_html][:id]
         end
 
-        label_options
+        label_options.except(*VIRTUAL_OPTIONS)
```

Comment:
Did you see why it is here in the first place? `html_options_for(:label)`  should only return things that are in the `label_html` key.